.github/workflows/increment_version.py

```diff
@@ -0,0 +1,106 @@
+"""A simple script to get current latest version from repo in working directory and
+increment it accordingly. This can also be used to alter the version in metadata
+files."""
+
+import argparse
+import json
+from os import environ
+from pathlib import Path
+import re
+import subprocess
+from typing import cast, Dict, Tuple, List
+import sys
+
+VERSION_TUPLE = Tuple[int, int, int]
+COMPONENTS = ["major", "minor", "patch"]
+
+
+def get_version() -> VERSION_TUPLE:
+    """get current latest tag and parse into a 3-tuple"""
+    result = subprocess.run(
+        ["gh", "release", "list", "--limit=1", "--json=tagName"],
+        capture_output=True,
+        check=True,
+    )
+    ver_json = cast(
+        List[Dict[str, str]], json.loads(result.stdout.decode(encoding="utf-8"))
+    )[0]
+    assert "tagName" in ver_json, f"got malformed output from gh-cli:\n{ver_json}"
+    ver_tag = ver_json["tagName"].lstrip("v")
+    print("Current version:", ver_tag)
+    ver_tuple = ver_tag.split(".")
+    assert len(ver_tuple) == 3, f"failed to parse version from tag {ver_tag}"
+    return int(ver_tuple[0]), int(ver_tuple[1]), int(ver_tuple[2])
+
+
+def increment_version(version: VERSION_TUPLE, bump: str = "patch") -> VERSION_TUPLE:
+    """Increment given version based on specified ``bump`` component."""
+    new_ver = list(version)  # make tuple mutable
+    component = COMPONENTS.index(bump)
+    new_ver[component] += 1
+    return tuple(new_ver)
+
+
+def update_metadata_files(version: str) -> bool:
+    """update the library metadata files with the new specified ``version``."""
+    made_changes = False
+
+    pio_meta_file = Path("library.json")
+    if pio_meta_file.exists():
+        pio_ver_pattern = re.compile(r'"version":\s+"\d+\.\d+\.\d+",')
+        data = pio_meta_file.read_text(encoding="utf-8")
+        data = pio_ver_pattern.sub(f'"version": "{version}",', data)
+        pio_meta_file.write_text(data, encoding="utf-8", newline="\n")
+        made_changes = True
+
+    arduino_meta_file = Path("library.properties")
+    if arduino_meta_file.exists():
+        arduino_ver_pattern = re.compile(r"version=\d+\.\d+\.\d+")
+        data = arduino_meta_file.read_text(encoding="utf-8")
+        data = arduino_ver_pattern.sub(f"version={version}", data)
+        arduino_meta_file.write_text(data, encoding="utf-8", newline="\n")
+        made_changes = True
+
+    return made_changes
```

Comment:
Should probably make sure that the file was actually changed. This would prevent problems pushing a commit from the CI where there are no files changes to push.